src/fidesops/util/logger.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import os
+from typing import Any, Mapping, Union
+
+MASKED = "MASKED"
+
+
+class NotPii(str):
+    """whitelist non pii data"""
+
+
+def get_fides_log_record_factory() -> Any:
+    """intercepts default LogRecord for custom handling of params"""
+
+    def factory(  # pylint: disable=R0913
+        name: str,
+        level: int,
+        fn: str,
+        lno: int,
+        msg: str,
+        args: Union[tuple[Any, ...], Mapping[str, Any]],
+        exc_info: Any,
+        func: str = None,
+        sinfo: str = None,
+    ) -> logging.LogRecord:
+        masked_args: tuple[Any, ...] = tuple(_mask_pii_for_logs(arg) for arg in args)
```

Comment:
This is a minor optimization, but this is going to be called _a lot_ . This calls a string comparison check with os.getenv == "True" on each parameter, be faster to do something like
```
 show_pii = os.getenv("LOG_PII") == "True"
 if not show_pii:
    ... mask ...
    ```